### PR TITLE
chore: update repo for api version 50.0

### PR DESCRIPTION
**Pull request by lcampos**
### What does this PR do?
Updates the repository to use v50.0 as the default api version

### What issues does this PR fix or reference?
@W-8205897@

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ commands:
             npx lerna version << pipeline.parameters.publish-type >> --force-publish --no-git-tag-version --exact --yes
             git add .
             export RELEASE_TAG="$(node -pe "require('./lerna.json').version")"
-            git commit -m "Updated version $RELEASE_TAG"
+            git commit -m "chore: updated version $RELEASE_TAG"
```
Comment (lcampos) [inline .circleci/config.yml:46]: @sfsholden updated the circleci commits to follow the new commitizen format

File: packages/plugin-templates/test/commands/force/analytics/template/create.test.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ describe('Analytics template creation tests:', () => {
             path.join('waveTemplates', 'foo', 'template-info.json'),
             {
               label: 'foo',
-              assetVersion: 49.0
+              assetVersion: 50.0
```
Comment (lcampos) [inline packages/plugin-templates/test/commands/force/analytics/template/create.test.ts:32]: At some point we should update the tests to read the version from `package.json`, but that will be on a different PR.

File: packages/plugin-templates/test/commands/force/analytics/template/create.test.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ describe('Analytics template creation tests:', () => {
             path.join('waveTemplates', 'foo', 'template-info.json'),
             {
               label: 'foo',
-              assetVersion: 49.0
```
Comment (smithgp) [inline packages/plugin-templates/test/commands/force/analytics/template/create.test.ts:32]: This should not be updated, nor the assetVersion in the template-info.json below. This number refers not the current version of the server, but to the format of the dashboard.json file, which is 49.0 -- the analytics server code uses this value to determine the parser & upgraders to use at runtime.
In analytics, It's normal to leave this assetVersion as-is until the format actually changes.

**Review by smithgp — CHANGES_REQUESTED** [review]

File: packages/plugin-templates/test/commands/force/analytics/template/create.test.ts (hunk 3 above)
Comment (lcampos) [reply packages/plugin-templates/test/commands/force/analytics/template/create.test.ts]: Cool, I've reverted this

**Review by smithgp — APPROVED** [review]

**Review by AnanyaJha — APPROVED** [review]

**Review by xyc — APPROVED** [review]
